addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -572,10 +572,11 @@ function makeActionManager(env) {
             const removeDialog = env.services.dialog.add(ActionDialog, actionDialogProps, {
                 onClose: () => {
                     if (dialog.remove) {
-                        if (dialog.onClose) {
-                            dialog.onClose();
-                        }
+                        const { onClose } = dialog;
                         dialog = {};
+                        if (onClose) {
+                            onClose({ special: true });
```

Comment:
what are the specs around this param actually?

Reply:
I don't know really. In legacy (14.0) ; dialog closed that way are not given anything (the on_close callback is passed jqueryEvent.info which seems undefined everytime).

I drew from the action_service:@DoActionButton where when we hit a "cancel" button we pass special to the callback. I don't think a "cancel" button is in any way different from the cross at the top right of a dialog....